tests/lib/nested.sh

```diff
@@ -23,41 +23,86 @@
 : "${NESTED_DISK_LOGICAL_BLOCK_SIZE:=512}"
 
 nested_wait_for_ssh() {
-    # TODO:UC20: the retry count should be lowered to something more reasonable.
-    local retry=800
-    local wait=1
+    local retry=${1:-800}
+    local wait=${2:-1}
 
-    until remote.exec "true"; do
+    until remote.exec "true" &>/dev/null; do
         retry=$(( retry - 1 ))
         if [ $retry -le 0 ]; then
-            echo "Timed out waiting for command 'true' to succeed. Aborting!"
             return 1
         fi
         sleep "$wait"
     done
 }
 
 nested_wait_for_no_ssh() {
-    local retry=200
-    local wait=1
+    local retry=${1:-200}
+    local wait=${2:-1}
 
-    while remote.exec "true"; do
+    while remote.exec "true" &>/dev/null; do
         retry=$(( retry - 1 ))
         if [ $retry -le 0 ]; then
-            echo "Timed out waiting for command 'true' to fail. Aborting!"
             return 1
         fi
         sleep "$wait"
     done
 }
 
+nested_wait_vm_ready() {
+    echo "Waiting the vm is ready to be used"
+    local retry=${1:-120}
+    local log_limit=${2:-30}
+
+    local output_lines=0
+    local serial_log="$NESTED_LOGS_DIR"/serial.log
+    while true; do
+        retry=$(( retry - 1 ))
+
+        # Check the timeout is reached
+        if [ $retry -le 0 ]; then
+            echo "Timed out waiting for vm ready. Aborting!"
+            return 1
+        fi
+
+        # Check the vm is active
+        if ! systemctl is-active "$NESTED_VM"; then
+            echo "Unit $nested_unit is not active. Aborting!"
+            return 1
+        fi
+
+        # Check during $limit seconds that the serial log is growing
+        # shellcheck disable=SC2016
+        retry -n "$log_limit" --wait 1 --quiet --env serial_log="$serial_log" --env output_lines="$output_lines" \
+            sh -c 'test "$(wc -l <"$serial_log")" -gt "$output_lines"'
+        output_lines="$(wc -l "$serial_log" | awk '{print $1;}')"
```

Comment:
If you use the input redirect then there's no need for awk
```suggestion
        output_lines="$(wc -l <"$serial_log")"
```